### PR TITLE
Add filtering for rent price

### LINKED ISSUE #102: Rent price filtering
(Split this ticket into multiple, for tracking. It's now
- #102
- #320 
- #321 
- #322 
- #324 )

This is blocked on how we store rent data, and how we want users to filter on it.

The table in the design doc has additional thoughts about how this filter might be structured:
https://app.box.com/integrations/googledss/openGoogleEditor?fileId=839071952572&trackingId=3#heading=h.gnp30ny3nbk3

### PR DESCRIPTION
**Pull request by abbiefarr**
## Issue

- Closes #102 

## Description

Adds ability to filter by rent price based on data in the units summary table.

UI:
image

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Prototype/POC (not to merge)
- [ ] This change is a refactor/address technical debt
- [ ] This change requires a documentation update
- [ ] This change requires a SQL Script

## How Can This Be Tested/Reviewed?

Performed some filtering a verified that the query was hitting the backend and the data was being filtered based on the units summary table.

NOTE: this is a little broken in certain cases, e.g. if you want to search for 2-bedrooms that are less than $600 a month, because the bedroom filter is looking at the units table, and the rent filter is looking at the summary table. We need way to generally determine which tables we should be joining to make queries. That's tracked in #530.

- [x] Desktop View
- [x] Mobile View

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [x] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have updated the changelog to include a description of my changes
- [x] I have run `yarn generate:client` if I made backend changes

### CODE REVIEW

File: backend/core/src/listings/dto/listing.dto.ts [hunk 1]
```diff
@@ -807,7 +807,25 @@ export class ListingFilterParams extends BaseFilter {
...
   })
-  [ListingFilterKeys.seniorHousing]?: boolean
+  [ListingFilterKeys.seniorHousing]?: boolean;
+
+  @Expose()
+  @ApiProperty({
+    type: Number,
+    example: "300",
+    required: false,
+  })
```
Comment (avaleske) [inline backend/core/src/listings/dto/listing.dto.ts:817]: Can you add `@IsOptional({ groups: [ValidationsGroupsEnum.default] })` above the `@IsNumberString...` here and below? As well as on senior housing, availability, and zipcode?

Upstream just merged changes that seem to make this validation work correctly, so I think we'll need to actually allow these to be optional now.

File: sites/public/pages/listings.tsx [hunk 2]
```diff
@@ -175,6 +181,28 @@ const ListingsPage = () => {
...
+              <Field
+                id="minRent"
+                name={ListingFilterKeys.minRent}
+                register={register}
+                type="number"
+                placeholder={t("t.min")}
+                prepend="$"
+                defaultValue={filterState?.minRent}
+              />
+              <div className="flex items-center">{t("t.to")}</div>
```
Comment (avaleske) [inline sites/public/pages/listings.tsx:195]: Smallest nit ever: when a user tabs onto the fields, the highlight kind of crowds the "to". Could we slightly increase the padding around it?

image

**Review by avaleske — APPROVED** [review]
LGTM! I had a few small nits, one of which is cause of some upstream changes that just happened after you mailed this PR.

**Review by willrlin — APPROVED** [review]

File: backend/core/src/listings/dto/listing.dto.ts (hunk 1 above)
Comment (abbiefarr) [reply backend/core/src/listings/dto/listing.dto.ts:820]: Done

File: sites/public/pages/listings.tsx (hunk 2 above)
Comment (abbiefarr) [reply sites/public/pages/listings.tsx]: Added padding: 
image

**Pull request merged**